flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/SchedulerNGFactoryFactoryTest.java

```diff
@@ -57,15 +63,28 @@ public void throwsExceptionIfSchedulerNameIsInvalid() {
         configuration.setString(JobManagerOptions.SCHEDULER.key(), "invalid-scheduler-name");
 
         try {
-            createSchedulerNGFactory(configuration);
+            createSchedulerNGFactory(configuration, testGraph);
         } catch (IllegalArgumentException e) {
             assertThat(
                     e.getMessage(),
                     containsString("Could not parse value 'invalid-scheduler-name'"));
         }
     }
 
-    private static SchedulerNGFactory createSchedulerNGFactory(final Configuration configuration) {
-        return SchedulerNGFactoryFactory.createSchedulerNGFactory(configuration);
+    @Test
+    public void fallBackIfBatchAndDeclarative() {
+        final Configuration configuration = new Configuration();
+        configuration.set(JobManagerOptions.SCHEDULER, JobManagerOptions.SchedulerType.Declarative);
+        final JobGraph jobGraph = new JobGraph("test");
+        jobGraph.setJobType(JobType.BATCH);
+        final SchedulerNGFactory schedulerNGFactory =
+                createSchedulerNGFactory(configuration, testGraph);
+
+        assertThat(schedulerNGFactory, is(instanceOf(DefaultSchedulerFactory.class)));
```

Comment:
I think we should instead check that it is not a factory for the declarative scheduler. This test should not care what the factory for the default scheduler is.